pkg/kubectl/resize.go

```diff
@@ -70,6 +83,19 @@ type ReplicationControllerResizer struct {
 	client.Interface
 }
 
+func CastResizeAsCondition(r Resizer, precondition *ResizePrecondition, namespace, name string, count uint) wait.ConditionFunc {
+	return func() (bool, error) {
+		switch _, err := r.Resize(namespace, name, precondition, count); err.(type) {
+		case nil:
+			return true, nil
+		case ControllerUpdateError:
```

Comment:
You could treat only PreconditionErrors as permanent failures and let it keep trying on all other errors; then you wouldn't have to make the ControllerUpdateError type.